backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -1158,24 +1168,131 @@ private String getStringFromLocationList(List<String> locationList) {
             .append(" ");
 
     StringBuilder joinClause = new StringBuilder();
+    String repoAlias = "repo";
+    joinClause.append(" INNER JOIN ").append(alias).append(".repository ").append(repoAlias);
     joinClause
         .append(" INNER JOIN ")
         .append(InternalFolderElementEntity.class.getSimpleName())
         .append(" folderElm ")
         .append(" ON ");
     joinClause.append("folderElm.folder_hash = ").append(alias).append(".rootSha ");
 
+    List<KeyValueQuery> predicates = request.getPredicatesList();
     List<String> whereClauseList = new ArrayList<>();
-    if (!request.getRepoIdsList().isEmpty()) {
-      whereClauseList.add(alias + ".repository.id IN (:repoIds) ");
-      parametersMap.put("repoIds", request.getRepoIdsList());
+    if (predicates != null && !predicates.isEmpty()) {
```

Comment:
Document different keys and their format this function is supporting please

Reply:
https://github.com/VertaAI/modeldb/pull/740/files#diff-ae50dfe67fc6d04ad1a941d85c9510ceR1157-R1167